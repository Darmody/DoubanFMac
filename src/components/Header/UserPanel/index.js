// @flow
import React, { PureComponent } from 'react'
import styled from 'styled-components'
import Arrow from 'components/Icons/Arrow'

const UserPanel = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`

const Avatar = styled.img`
  height: 1.375rem;
  width: 1.375rem;
  border-radius: 50%;
  margin-right: 1rem;
`

const ListenedWrapper = styled.span`
  height: 1.125rem;
  line-height: 1.125rem;
  padding: 0 0.188rem;
  border-radius: 2px;
  font-size: 11px;
  font-weight: 400;
  text-align: center;
  display: inline-block;
  background-color: transparent;
  border: 1px solid rgb(142, 217, 156);
  transform: skew(-10deg);
`

const Listened = styled.span`
  display: inline-block;
  color: rgb(47, 152, 66);
  transform: skew(10deg);
`

const indicatedArrowCss = `
  cursor: pointer;
  transition: all 0.3s ease-out;
  &:hover {
    transform: rotate(180deg);
  }
`

type Props = {
  me: Object,
}

export default class UserPanelComponent extends PureComponent {
  props: Props

  render() {
    const { me } = this.props

    return (
      <UserPanel>
        <Avatar src={me.icon} alt="" />
        <ListenedWrapper>
          <Listened>已听{me.playedNum}首</Listened>
        </ListenedWrapper>
        <Arrow styles={indicatedArrowCss} />
      </UserPanel>
    )
  }
}
