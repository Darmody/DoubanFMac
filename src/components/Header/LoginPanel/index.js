// @flow
import React, { PureComponent } from 'react'
import styled from 'styled-components'
import Modal from 'react-modal'
import { LoginForm as LoginFormContainer } from 'containers'
import Form from './Form'

const LoginPanel = styled.a`
  font-size: .75rem;
  margin-left: .625rem;
  line-height: 1.5rem;
  color: #8f8e94;
  text-decoration: none;
  outline: none;
  cursor: pointer;
}
`

const modalStyle = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
  }
}

type Props = {
}

type State = {
  modalIsOpen: boolean,
}

export default class LoginPanelContainer extends PureComponent {
  constructor() {
    super()

    this.state = {
      modalIsOpen: false,
    }
  }

  state: State
  props: Props

  openModal = () => {
    this.setState({ modalIsOpen: true })
  }

  closeModal = () => {
    this.setState({ modalIsOpen: false })
  }

  render() {
    return (
      <div>
        <Modal
          isOpen={this.state.modalIsOpen}
          onRequestClose={this.closeModal}
          contentLabel="登录"
          style={modalStyle}
        >
          <LoginFormContainer component={Form} closeModal={this.closeModal} />
        </Modal>
        <LoginPanel
          href="javascript:void(0);"
          onClick={this.openModal}
        >
          登录
        </LoginPanel>
      </div>
    )
  }
}
