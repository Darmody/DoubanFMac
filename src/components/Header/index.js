// @flow
import React, { PureComponent } from 'react'
import styled from 'styled-components'
import UserPanel from './UserPanel'
import LoginPanel from './LoginPanel'
import LogoImage from './logo.svg'

const HeaderWrapper = styled.div`
  width: 100%;
`

const Header = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 45.975rem;
  margin: 1.75rem auto;
`

const FrontPanel = styled.div`
  display: flex;
  align-items: flex-start;
`

const Logo = styled.img`
  height: 2.5rem;
  width: 6.25rem;
`

const FeedBack = styled.a`
  font-size: 12px;
  color: #8f8e94;
  text-decoration: none;
  cursor: pointer;
  margin: 0.188rem 0 0 2.5rem;
`

export default class HeaderContainer extends PureComponent {
  props: {
    auth: boolean,
  }

  renderUserPanel = () => {
    if (this.props.auth) {
      return <UserPanel />
    }

    return <LoginPanel />
  }
  render() {
    return (
      <HeaderWrapper>
        <Header>
          <FrontPanel>
            <Logo src={LogoImage} alt="" />
            <FeedBack href="javascript:void(0);">我要反馈</FeedBack>
          </FrontPanel>
          {this.renderUserPanel()}
        </Header>
      </HeaderWrapper>
    )
  }
}
