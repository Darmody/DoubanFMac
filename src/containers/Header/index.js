// @flow
import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import { Header } from 'components'

type Props = {
  auth: boolean,
}

class HeaderContainer extends PureComponent {
  props: Props

  render() {
    return (
      <Header auth={this.props.auth} />
    )
  }
}

export default connect(
  state => ({
    auth: state.auth.id,
  })
)(HeaderContainer)
