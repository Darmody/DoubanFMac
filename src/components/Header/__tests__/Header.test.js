import React from 'react'
import renderer from 'react-test-renderer'
import Header from 'components/Header'

describe('Component::Header', () => {
  test('auth = false 时渲染 LoginPanel', () => {
    const component = renderer.create(<Header />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('auth = true 时渲染 UserPanel', () => {
    const component = renderer.create(<Header auth />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

})
