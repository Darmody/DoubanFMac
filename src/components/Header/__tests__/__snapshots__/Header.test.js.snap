// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::Header auth = false 时渲染 LoginPanel 1`] = `
<div
  className="bMAZFf"
>
  <div
    className="jkBogs"
  >
    <div
      className="glkAad"
    >
      <img
        alt=""
        className="ijNzyI"
        src={Object {}}
      />
      <a
        className="ipowgx"
        href="javascript:void(0);"
      >
        我要反馈
      </a>
    </div>
    <div>
      <noscript />
      <a
        className="fYxbAV"
        href="javascript:void(0);"
        onClick={[Function]}
      >
        登录
      </a>
    </div>
  </div>
</div>
`;

exports[`Component::Header auth = true 时渲染 UserPanel 1`] = `
<div
  className="bMAZFf"
>
  <div
    className="jkBogs"
  >
    <div
      className="glkAad"
    >
      <img
        alt=""
        className="ijNzyI"
        src={Object {}}
      />
      <a
        className="ipowgx"
        href="javascript:void(0);"
      >
        我要反馈
      </a>
    </div>
    <div
      className="ePLfcs"
    >
      <img
        alt=""
        className="kjYOjA"
        src={Object {}}
      />
      <span
        className="gHMsaP"
      >
        <span
          className="hpWlLF"
        >
          已听15606首
        </span>
      </span>
      <svg
        className="bNDeOQ"
        height="16"
        title="Arrow"
        viewBox="0 0 16 16"
        width="16"
      >
        <desc>
          Icon
        </desc>
        <g
          fill="none"
          fillRule="evenodd"
          id="Page-1"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="#c6c6c6"
            id="Triangle-1"
            points="8 6 12 11 4 11 "
            transform="translate(8.000000, 8.500000) scale(1, -1) translate(-8.000000, -8.500000) "
          />
        </g>
      </svg>
    </div>
  </div>
</div>
`;
